runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java

```diff
@@ -621,25 +617,16 @@ private void takeScreenshot(String fileName, Image image)
 		// Draw the game onto the screenshot
 		graphics.drawImage(image, gameOffsetX, gameOffsetY, null);
 
-		File playerFolder;
-		if (client.getLocalPlayer() != null && client.getLocalPlayer().getName() != null)
+		File playerFolder = SCREENSHOT_DIR;
+		if (!Strings.isNullOrEmpty(client.getUsername()))
```

Comment:
Could you check the behavior of `getUsername()`? From what I read in the docs, if the scenario was:

username `bob@bobmcbobface.com` (post 2011?) or `bobbyplayer` (pre 2011?)
password `redacted`
in game display name `Bob McBobFace`

Then you'd get back `bob@bobmcbobface.com` or `bobbyplayer`, assuming I understand the docs correctly.

It looks like this would inadvertently expose an email address or a username. I know I don't want my username floating around my file system given it's sensitive information, especially when it comes to shared computers. Prefer reverting back to old behavior for this conditional?

Reply:
Be sure to change line 623 below as necessary.

Reply:
Well, the issue with current approach is that if you change your display name in-game, then screenshot will start saving in different folder, and if you for some reason pick display name that you were using on other account before, screenshot plugin will start overwriting those, effectively deleting your data.

Reply:
Yeah, they both sound like hairy situations. But, how will the file get overwritten? Time will be part of the file name, and if (somehow, because dumb luck with time and/or clock skew) time was identical, a _1 or _2 would be appended?

Reply:
Well, with this patch they will not be overwriten, on master they are. But still that do not solves the issue of different accounts being mixed in 1 screenshot folder, that is supposed to be unique per-account.

Reply:
I feel you, I just don't like exposing username. Another scenario I had in mind is, you know how all the Twitch streamers hide their username and have the client auto remember so they never show it on stream? Maybe one day if they want to show a screenshot from their screenshot archive to the stream, the username will be visible in all its glory. That would not be ideal.